xdsl/interpreters/experimental/wgsl_printer.py

```diff
@@ -0,0 +1,248 @@
+from __future__ import annotations
+
+from functools import singledispatchmethod
+from typing import IO, cast
+
+from xdsl.dialects import arith, builtin, gpu, memref
+from xdsl.dialects.memref import MemRefType
+from xdsl.ir import Operation, SSAValue
+from xdsl.ir.core import Attribute
+
+
+class WGSLPrinter:
+    name_dict: dict[SSAValue, str] = dict()
+    count = 0
+
+    def wgsl_name(self, v: SSAValue):
+        if v not in self.name_dict.keys():
```

Comment:
You can drop the `.keys()` here, `x in <some dict>` already checks if x is a key of the dict.